File: lang/codes/manage_zone/app/routers/zone.py
# @Time:        2022-08-04
# @Author:      Sseve
# @File:        zone.py
# @Description: all api of zone

from fastapi import APIRouter, Depends
from internal.models import ZoneAdd
from internal.models import ZoneManage
from internal.db.zone import ZoneDb
from internal.utils import FabricHandle, get_current_user


router = APIRouter(prefix="/zone/api", tags=["zone api"])


@router.post("/zone")
async def zone(zone_list: ZoneAdd, _ = Depends(get_current_user)):
    """
    Args:
        @add => zone arguments like:
            {
                "zones": [
                    {"name":"xxx", "ip":"xxx", "zone_id":1},
                    {"name":"xxx", "ip":"xxx", "zone_id":2}
                ]
            }
    Return:
    """
    # zone info into db
    if not await ZoneDb(zones=zone_list.zones).add:
        return {"code": 400, "msg": "add zone to db failed"}
    zones = [await FabricHandle(zone=zone).add_zone for zone in zone_list.zones]
    print(zones)
    


@router.put("/zone")
async def zone(manage: ZoneManage, _ = Depends(get_current_user)):
    """
    Args:
        @target => action of zone like: [con|bin|start|stop|check|reload]
        @zones => zone arguments like:
            {
                "target":"con",
                "zones": [
                    {"name":"xxx","ip":"xxx", "id":1},
                    {"name":"xxx","ip":"xxx", "id":2}
                ]
            }
    Return:
    """
    print(manage)

File: lang/codes/bnzt-web/src/api/zone.js
import request from '@/utils/request'


// 区服列表
export function ZoneList() {
    return request({
        url: "/zone/api/zones",
        method: "get"
    })
}

// 添加区服
export function AddZone(data) {
    return request({
        url: "/zone/api/zones",
        method: "post",
        data: data,
        timeout: 600000
    })
}

// 区服操作
export function ManZone(data) {
    return request({
        url: "/zone/api/man",
        method: "post",
        data: data,
        timeout: 600000
    })
}

// 获取主机资源
export function GetHostData(data) {
    return request({
        url: "/zone/api/host",
        method: "post",
        data: data,
        timeout: 600000
    })
}